plugins/ksp/src/org/jetbrains/kotlin/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -201,6 +199,15 @@ class ResolverImpl(
         } as FunctionDescriptor?
     }
 
+    fun resolvePropertyDeclaration(property: KSPropertyDeclaration): PropertyDescriptor? {
+        return when (property) {
+            is KSPropertyDeclarationImpl -> resolveDeclaration(property.ktProperty)
+            is KSPropertyDeclarationDescriptorImpl -> property.descriptor
+            is KSPropertyDeclarationJavaImpl -> resolveJavaDeclaration(property.psi)
```

Comment:
You don't need to check override for symbols from Java, Java only has fields and therefore property override does not apply.

Same for other occurrences of Java symbols related logic.

Reply:
Hm, not sure I completely follow. What does `KSPropertyDeclarationJavaImpl` represent then?

Reply:
If you look at the constructor of `KSPropertyDeclarationJavaImpl` you will find it is taking a `PsiField`. While there is no property in Java, since KSP is Kotlin based, when modeling Java we are still trying best to model it in Kotlin way.